substrate/frame/balances/src/lib.rs

```diff
@@ -768,12 +769,19 @@ pub mod pallet {
 						"account with a non-zero reserve balance has no provider refs, account_id: '{:?}'.",
 						who
 					);
-					a.free = a.free.max(Self::ed());
+					a.free = a.free.max(ed);
```

Comment:
Btw would this not mess up the `TotalIssuance`?

Reply:
It should not as it is now it just move funds from reserved to free